src/test/java/com/microsoft/sqlserver/jdbc/connection/PoolingTest.java

```diff
@@ -139,4 +145,52 @@ public void testConnectionPoolClientConnectionId() throws SQLException {
 
         assertEquals(Id1, Id2, "ClientConnection Ids from pool are not the same.");
     }
+    
+    @Test
+    public void testHikariCP() throws SQLException {
+        HikariConfig config = new HikariConfig();
+        config.setJdbcUrl(connectionString);
+        HikariDataSource ds = new HikariDataSource(config);
+
+        connect(ds);
+    }
+
+    @Test
+    public void testApacheDBCP() throws SQLException {
+        BasicDataSource ds = new BasicDataSource();
+        ds.setUrl(connectionString);
+
+        connect(ds);
+    }
+
+    private static void connect(DataSource ds) throws SQLException {
+        Connection con = null;
+        PreparedStatement pst = null;
+        ResultSet rs = null;
+
+        try {
+            con = ds.getConnection();
+            con.isValid(5);
+
+            pst = con.prepareStatement("SELECT SUSER_SNAME()");
```

Comment:
hello @fwi , could you please inform me what the benefits are by testing setQueryTimeout() instead of isValid()? from my understanding, both create a new thread for the timeout timer